### DISCUSSION TITLE
Log file rotation

### DISCUSSION THREAD

**Discussion by yamadapc** [vercel/turborepo#7298] [Help] [answered]
## Behaviour

Running turborepo on a large repository generated a 180GB log file on `~/.local/share/turborepo/logs`

Most entries are:

```
2024-02-07T00:48:22.020579Z TRACE turborepo_filewatch: manually watching [...]
```

## Expected behaviour

Log file is rotated and has a reasonable size size limit

**Comment by arlyon** [comment]
Hey, having all those trace things without opting in should not be happening. I would like to first make sure that we have more sane defaults, and then rotate based on the size (as well as the current behaviour which is per-date-per-repo).

**Reply by yamadapc** [reply]
Makes sense ; maybe something around - https://github.com/vercel/turbo/blob/4e5ae8dc108c895c31d17e575c667f2fc5b0d3c1/crates/turborepo-lib/src/tracing.rs#L129

**Reply by arlyon** [reply, marked as answer]
This will be (mostly) fixed by https://github.com/vercel/turbo/pull/7526. It dramatically reduces the log output (we had accidentally been logging span events which are very verbose) as well as allowing cleanup using `turbo clean`
